app/scripts/inpage.js

```diff
@@ -121,3 +121,69 @@ function restoreContextAfterImports () {
     console.warn('MetaMask - global.define could not be overwritten.')
   }
 }
+
+/* ------ HANDLE INJECTING PLUGINS ------- */
+
+var plugins = {};
+
+try {
+var rand = function(){ return Math.ceil(Math.random()*10000) }
+// the following should be dynamic, NOT hard coded,
+// based on a UI a user can use to add directories or URLs.
+// We DO NOT want to have to add dependencies to MetaMask for plugins,
+// but for now, I did, just to get the gulp build system working properly.
+plugins.GUN = require('gun/gun.min.js');
+plugins.SEA = require('gun/sea.js');
+
+// NOTE: SEA should actually be core, not considered a plugin!
+// It is what other plugins would use as their metamask crypto API.
```

Comment:
That's a good note. Distinguishing the crypto core of SEA vs the db layer of Gun.

For other reviewers, this is building on a conversation we've had about how could we build towards allowing GUN integration without:
- Making MetaMask a political "king maker".
- Bringing the MetaMask bundle size over what Firefox will permit.
- Allowing users to opt-in to the additional networks they're interested in.

This had us seeking a minimal "plugin" interface that would allow users to add GUN as a plugin. Here we're pointing out that the SEA library is more about identity/cryptography than GUN, which is the peer/gossip layer, if I understand correctly.

Reply:
👍